### PR TITLE
Revert params.fetch to params.expect conversions in scaffold

### PR DESCRIPTION
**Pull request by martinemde**
### Motivation / Background

 In rails/jbuilder#573 we stumbled on the case where the scaffold was generating `params.expect(table_name: {})` which has the unintended consequence of requiring that a param is given when there are no expected params.

### Detail

This PR reverts 2 small changes to the scaffold templates introduced by #51674 so that default scaffolds don't return 400 Bad Request upon creation with no attributes.

### Additional information

Thanks to @jeromedalbert for noticing that I had made the very changes that I pointed out as problematic in his PR in jbuilder. Teamwork!

### Checklist

Before submitting the PR make sure the following are checked:

* [X] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [X] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [ ] Tests are added or updated if you fix a bug or add a feature.
* [ ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

**Comment by jeromedalbert** [comment]
You need something like [this](https://github.com/jeromedalbert/rails/commit/e5119cf90b34f349160ad2b6f15d257e18ab61ad.patch) to fix tests

**Comment by martinemde** [comment]
> I think you need something like this to fix tests.

Yes, of course. Thank you again, @jeromedalbert. That's what I get for coding in a hurry between other things.

Updated and tested locally. Should be good now.

**Comment by martinemde** [comment]
Updated again to add a test that was missing for api controller generation.

**Comment by hachi8833** [comment]
Current edgeguides suggest us to use `expect`, and the descriptions of `#fetch` method for strong parameters has been removed from current edgeguides:

https://edgeguides.rubyonrails.org/action_controller_overview.html#strong-parameters

But the template for controller has been updated to use `fetch` in the PR. I'm confused with the inconsistency between edgeguide and the generated controller. Which methods should we use primarily -- `fetch` and `expect`?

@martinemde do you have any thoughts to resolve the inconsistency?

**Comment by martinemde** [comment]
@hachi8833 `fetch` serves a very specific purpose of returning a value that you aren't certain will be there. `fetch` is like the opposite of `expect` and should be used in actions where you expect the request to be made without that param at least some of the time.

The reason it was reverted here is that `fetch` handles the case where there are no params given, returning an empty hash. If `expect` was used here then the freshly generated scaffold would fail when no attributes were given.

tl;dr: Always use `expect` unless the params are optional. (or, simply, use expect unless it doesn't do what you want)

**Comment by hachi8833** [comment]
@martinemde I got it, thank you! 😂

**Comment by martinemde** [comment]
@hachi8833 awesome, thanks for asking. This revert here was actually me catching a mistake in my first PR and fixing it real quick. 😄

**Comment by martinemde** [comment]
@hachi8833 If you're interested, I wrote an article about [how to use `params.expect`](https://martinemde.com/2024/10/22/how-to-rails-params-expect.html). It's even more wordy and verbose than my replies here 😆
